### DISCUSSION TITLE
New metric: largest message size in a queue

### DISCUSSION THREAD

**Discussion by johanrhodin** [rabbitmq/rabbitmq-server#2754] [Ideas]
The HTTP API for `/api/queues/vhost/name` could have a metric for the largest message in the queue.
Name suggestion: "message_bytes_largest_message".

We've developed in-house "trick" based methods to find it, but would be good to have a standardized way to get it from the API and display in the management interface https://server-name/#/queues/vhost/name.

Other users requesting something along the same line: 
- https://groups.google.com/g/rabbitmq-users/c/aIR6bbX1cwo/m/gNHFZ5cfAQAJ
- https://groups.google.com/g/rabbitmq-users/c/Gjbte49h4g4/m/abAt-feFBAAJ

**Comment by lukebakken** [comment]
> We've developed in-house "trick" based methods to find it

It would be great to see a pull request with an implementation. Finding the largest message requires a queue traversal which could potentially be expensive. Tracking the largest current message could be potentially complex.

**Reply by johanrhodin** [reply]
Yeah we don't have an erlang implementation, we just use Linux tools to inspect the file size of the .idx and .rdq files. Not perfect by any means, but at least gives us a hint of what the largest message can be.
